src/main/xerus/monstercat/tabs/TabSettings.kt

```diff
@@ -224,4 +237,33 @@ class TabSettings: VTab() {
 		data class Feedback(val subject: String, val message: String)
 	}
 	
+	enum class PriorityList(val priorities: List<String>){
+		SGL_ALB_COL(listOf(SINGLE, ALBUM, MCOLLECTION, BESTOF, MIXES, PODCAST)),
+		ALB_SGL_COL(listOf(ALBUM, SINGLE, MCOLLECTION, BESTOF, MIXES, PODCAST)),
+		COL_SGL_ALB(listOf(MCOLLECTION, SINGLE, ALBUM, BESTOF, MIXES, PODCAST)),
+		COL_ALB_SGL(listOf(MCOLLECTION, ALBUM, SINGLE, BESTOF, MIXES, PODCAST));
+		
+		companion object {
+			fun findFromString(string: String): PriorityList{
+				PriorityList.values().forEach {
+					if (string == getString(it))
+						return it
+				}
+				return SGL_ALB_COL
+			}
+			
+			fun findFromList(list: List<String>): PriorityList{
+				PriorityList.values().forEach {
```

Comment:
For these methods, you can use the `find` function instead of forEach, and then add the default value via `?:`. That way you could even write the function in expression syntax.

Reply:
5773bcb0210f66cd9a75b2182376218ba739f728